portality/bll/services/query.py

```diff
@@ -80,34 +80,43 @@ def _post_filter_search_results(self, cfg, res):
 
         return res
 
-    def search(self, domain, index_type, raw_query, account, additional_parameters):
+    def _get_query(self, cfg, raw_query):
         query = Query()
         if raw_query is not None:
             query = Query(raw_query)
 
-        cfg = self._get_config_for_search(domain, index_type, account)
-
-        # check that the request values permit a query to this endpoint
-        required_parameters = cfg.get("required_parameters")
-        if required_parameters is not None:
-            for k, vs in required_parameters.iteritems():
-                val = additional_parameters.get(k)
-                if val is None or val not in vs:
-                    raise exceptions.AuthoriseException()
-
         # validate the query, to make sure it is of a permitted form
         if not self._validate_query(cfg, query):
             raise exceptions.AuthoriseException()
 
+        # add any required filters to the query
+        query = self._pre_filter_search_query(cfg, query)
+        return query
+
+    def _get_dao_klass(self, cfg):
         # get the name of the model that will handle this query, and then look up
         # the class that will handle it
         dao_name = cfg.get("dao")
         dao_klass = plugin.load_class(dao_name)
         if dao_klass is None:
             raise exceptions.NoSuchObjectException(dao_name)
+        return dao_klass
 
-        # add any required filters to the query
-        query = self._pre_filter_search_query(cfg, query)
+    def search(self, domain, index_type, raw_query, account, additional_parameters):
```

Comment:
as per my comment below, I think we should change this to have `additional_parameters=None` so that we can bypass parameter validation if needed (which it would be for the API)